### DISCUSSION TITLE
CSS selector - variation to select elements in table (loop)

### DISCUSSION THREAD

**Discussion by JehanO** [AutomaApp/automa#1753] [Q&A]
Hi all,

It might be a dumb and very newbie question, but I have the following CSS selectors on a website, and want to loop them all:
`.css-5ef180 > .list-reset:nth-child(1) > .border`
`.css-5ef180 > .list-reset:nth-child(2) > .border`

The idea would be to say n should variate from 1 to xx in order to select them all.
I have set up a loop data from 1 to 10, followed by a click-element: 
`.css-5ef180 > .list-reset:nth-child({{loopData.ETns_g.$index}}) > .border`

How should I do to make it work?

Thanks!
Best

PS: in case of any need, here is the website I want to automate things: [sortlist.fr](https://www.sortlist.fr/seo/paris-fr)
<img width="804" alt="Capture d’écran 2024-06-05 à 12 17 41" src="https://github.com/AutomaApp/automa/assets/57331591/1fe5812a-57bc-4c15-a39f-d5c3965ee81d">
And this is the kind of data I want to click-on. (make sure you're in list view and not cards view to have the right css selector)

**Comment by jingbof** [comment]
I believe you're on the right track. Instead of trying to select them all and then loop one by one, I’d rather loop from 1 to 1000 (a really big number) and check if the element exists on the page. If it doesn’t, just skip it. The check should be fairly quick.

**Comment by happyTonakai** [comment]
Hello! Did you solve the problem? I face the same question. I set a loop data (loop through numbers from 1 to 48) and then connect to a click element, the CSS selector is like yours, followed by a delay and then a loop breakpoint. But it seems not working at all..
